modules/programs/yazi.nix

```
@@ -199,9 +199,41 @@ in {
       };
       "yazi/init.lua" = mkIf (cfg.initLua != null) { source = cfg.initLua; };
     } // (mapAttrs'
-      (name: value: nameValuePair "yazi/plugins/${name}" { source = value; })
-      cfg.plugins) // (mapAttrs'
-        (name: value: nameValuePair "yazi/flavors/${name}" { source = value; })
-        cfg.flavors);
+      (name: value: nameValuePair "yazi/flavors/${name}" { source = value; })
+      cfg.flavors) // (let
+        # Make sure that the directory ends in `.yazi`, to comply with specification.
+        # `pluginName` is essential, it's needed to apply config in yazi's `init.lua`
+        ensureSuffix = pluginName:
+          if lib.hasSuffix ".yazi" pluginName then
+            "yazi/plugins/${pluginName}"
+          else
+            "yazi/plugins/${pluginName}.yazi";
```

```suggestion
            "yazi/plugins/${pluginName}.yazi";
```

`foo.yazi` is the full package name, while `foo` is the plugin name.

It's best to only support the plugin name to stay consistent with Yazi's behavior. For example, use the `plugin` command as `plugin foo` instead of `plugin foo.yazi`. Yazi's package manager also uses `use = "owner/foo"` instead of `use = "owner/foo.yazi"`.

---

I agree, but it's a breaking change, so I'd also like to add at least a warning if a pluginName ends in `.yazi`.

btw, thanks for your work on yazi ^^